src/rewriter.coffee

```diff
@@ -649,7 +650,8 @@ exports.Rewriter = class Rewriter
     condition = (token, i) ->
       [tag] = token
       [prevTag] = @tokens[i - 1]
-      tag is 'TERMINATOR' or (tag is 'INDENT' and prevTag not in SINGLE_LINERS)
+      [nextTag] = @tokens[i + 1] unless i is @tokens.length - 1
+      tag is 'TERMINATOR' and nextTag not in LEADING_LOGICAL or (tag is 'INDENT' and prevTag not in SINGLE_LINERS)
```

Comment:
@GeoffreyBooth made a small update to support leading logical operators inside eg an `if` condition (added corresponding tests). For example:
```
if someCondition
    and someOtherCondition
    or somethingElse
  doSomething()
```
Basically what was happening was since `tagPostfixConditionals()` runs before `tagLeadingLogical()`, the `IF` was seeing a following `TERMINATOR` (which hadn't yet been converted into eg `LEADING_AND`) and deciding that it was a `POST_IF`

Can't just move `tagLeadingLogical()` to before `tagPostfixConditionals()`, since there's a dependency: `tagLeadingLogical()` should run after `addImplicitBracesAndParens()`, which should run after `tagPostfixConditionals()`

So basically detect what will become an eg `LEADING_AND` here to avoid tagging as a postfix conditional